File: src/main/java/com/googongill/aditory/service/dto/category/UpdateCategoryResult.java
package com.googongill.aditory.service.dto.category;

import com.googongill.aditory.domain.Category;
import com.googongill.aditory.domain.enums.CategoryState;
import lombok.Builder;
import lombok.Getter;

import java.time.LocalDateTime;

@Getter
@Builder
public class UpdateCategoryResult {
    private Long categoryId;
    private String categoryName;
    private String asCategoryName;
    private CategoryState categoryState;
    private LocalDateTime createdAt;
    private LocalDateTime lastModifiedAt;

    public static UpdateCategoryResult of(Category category) {
        return UpdateCategoryResult.builder()
                .categoryId(category.getId())
                .categoryName(category.getCategoryName())
                .asCategoryName(category.getAsCategoryName())
                .categoryState(category.getCategoryState())
                .createdAt(category.getCreatedAt())
                .lastModifiedAt(category.getLastModifiedAt())
                .build();
    }
}

File: src/main/java/com/googongill/aditory/service/dto/category/UpdateCategoryResult.kt
package com.googongill.aditory.service.dto.category

import com.googongill.aditory.domain.Category
import com.googongill.aditory.domain.enums.CategoryState
import java.time.LocalDateTime

data class UpdateCategoryResult(
    val categoryId: Long?,
    val categoryName: String,
    val asCategoryName: String,
    val categoryState: CategoryState,
    val createdAt: LocalDateTime?,
    val lastModifiedAt: LocalDateTime?
) {

    companion object {
        fun of(category: Category): UpdateCategoryResult {
            return UpdateCategoryResult(
                categoryId = category.id,
                categoryName = category.categoryName,
                asCategoryName = category.asCategoryName,
                categoryState = category.categoryState,
                createdAt = category.createdAt,
                lastModifiedAt = category.lastModifiedAt
            )
        }
    }
}
